packages/client/components/PokerEstimateHeaderCard.tsx

```diff
@@ -4,12 +4,58 @@ import {createFragmentContainer} from 'react-relay'
 import {PokerEstimateHeaderCard_stage} from '../__generated__/PokerEstimateHeaderCard_stage.graphql'
 import PokerEstimateHeaderCardContent from './PokerEstimateHeaderCardContent'
 import PokerEstimateHeaderCardError from './PokerEstimateHeaderCardError'
-import PokerEstimateHeaderCardGitHub from './PokerEstimateHeaderCardGitHub'
 import PokerEstimateHeaderCardParabol from './PokerEstimateHeaderCardParabol'
 
 interface Props {
   stage: PokerEstimateHeaderCard_stage
 }
+
+type Integration = NonNullable<NonNullable<PokerEstimateHeaderCard_stage['task']>['integration']>
+export type IntegrationHeaderFields = {
+  cardTitle: string
+  descriptionHTML: string
+  url: string
+  linkTitle: string
+  linkText: string
+}
+
+const getHeaderFields = (integration: Integration | null): IntegrationHeaderFields | null => {
+  if (!integration) return null
+  const {__typename} = integration
+  switch (__typename) {
+    case 'JiraServerIssue':
+    case 'JiraIssue':
+      const name = __typename === 'JiraIssue' ? 'Jira' : 'Jira Server'
+      const {summary, descriptionHTML, jiraUrl, issueKey} = integration
+      return {
+        cardTitle: summary,
+        descriptionHTML,
+        url: jiraUrl,
+        linkTitle: `${name} Issue #${issueKey}`,
+        linkText: issueKey
+      }
+    case '_xGitHubIssue':
+      const {number, title: githubTitle, bodyHTML, url} = integration
+      return {
+        cardTitle: githubTitle,
+        descriptionHTML: bodyHTML,
+        url,
+        linkTitle: `GitHub Issue #${number}`,
+        linkText: `#${number}`
+      }
+    case '_xGitLabIssue':
+      const {iid, title, descriptionHtml, webUrl} = integration
+      return {
+        cardTitle: title,
+        descriptionHTML: descriptionHtml || '',
```

Comment:
Is there a reason to use `||` instead of `??` ?

Reply:
We could use either one. `||` will use the right-hand side of the operation if the left-hand side is falsy, e.g. `0`, whereas `??` will only do this if it is null or undefined. Here, we're checking whether `descriptionHtml` is a string or null, so `??` is probably more descriptive. I've pushed an update changing this 👍